src/main/java/dataAnalyzer/TriangleFormationAnalyzer.java

```diff
@@ -10,38 +10,32 @@ public static boolean[] analyze(List<Price> priceList) {
         boolean[] triangleformations = new boolean[priceList.size()];
         double[] blurredValues = Blur.averageBlur(priceList);
 
-        for (int i = 0; i < triangleformations.length; i++) {
-            double lastPrice = blurredValues[i];
-
-            // Check for decreasing formation
-            int formationLen = 0;
-            for (int j = i + 1; j < triangleformations.length; j++, formationLen++) {
-                double currentPrice = blurredValues[j];
-                if (currentPrice > lastPrice)
-                    break;
-                lastPrice = currentPrice;
-            }
-
-            // Check for increasing formation
-            if (formationLen == 0) {
-                for (int j = i + 1; j < triangleformations.length; j++, formationLen++) {
-                    double currentPrice = blurredValues[j];
-                    if (currentPrice < lastPrice)
-                        {break;}
-                    lastPrice = currentPrice;
-                }
-            }
-
-            if (formationLen >= MIN_FORMATION_LEN) {
-                for (int k = i; k < i + formationLen; k++) {
-                    triangleformations[k] = true;
-                }
-                i += formationLen;
-            } else {
-                triangleformations[i] = false;
-            }
+        double compDif = Math.abs(blurredValues[0] - blurredValues[1]);
+    double maxDifference = 1.0;
+    int formationLen = 0;
+    boolean end = false;
+
+    for (int i = 2; i < blurredValues.length; i++) {
+      double currentDifference = Math.abs(blurredValues[i] - blurredValues[i - 1]);
+      if (currentDifference > maxDifference && currentDifference < compDif) {
+        formationLen++;
+      } else if (currentDifference <= maxDifference && currentDifference < compDif) {
+        end = true;
+      }
+      if (end&&MIN_FORMATION_LEN<=formationLen) {
+        for (int j = 0; j <= formationLen; j++) {
```

Comment:
1. Shouldn't this be `<` instead of `<=`, because you start counting at 0 and want to have `formationLen` many iteration, right? Or do you actually want `formationLen + 1` many iterations?
2. Instead of setting `triangleFormations[i - j - 1]` and `triangleFormations[i - j - 2]` to `true` (which in most iterations would change nothing and just cost time), you could instead set `triangleFormations[i - formationsLen]` and ` triangleFormations[i -1 - formationsLen]` to true outside the loop. Or arguably even nicer - you could just iterate up to `j < formationLen + 2` and then you only need a single line in the loop

Reply:
1. I actually want formationLen+1 iterations. Its because I didn't add another count when i set` end=true`.
2. good point

Reply:
1. ah that makes sense 👍 I just get anxious anytime I see a `<=` in the condition of a for-loop xD
2. 👍

Reply:
1. so fair
2. fixed it